src/rules/enforce-querykey-ts.ts

```
@@ -0,0 +1,288 @@
+            // Check if the first argument is an object expression
+            if (firstArg.type === AST_NODE_TYPES.ObjectExpression) {
+              // Find the key property in the object
+              const keyProperty = firstArg.properties.find(
+                (prop): prop is TSESTree.Property =>
+                  prop.type === AST_NODE_TYPES.Property &&
+                  prop.key.type === AST_NODE_TYPES.Identifier &&
+                  prop.key.name === 'key',
+              );
+
+              // If key property exists, check its value
+              if (keyProperty && keyProperty.value) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)